packages/@aws-cdk/core/lib/app.ts

```diff
@@ -112,7 +136,11 @@ export class App extends Construct {
 
     // both are reverse logic
     this.runtimeInfo = this.node.tryGetContext(cxapi.DISABLE_VERSION_REPORTING) ? false : true;
-    this.outdir = props.outdir || process.env[cxapi.OUTDIR_ENV];
+
+    // Need to determine fixed output directory already, because we must know where
+    // to write sub-assemblies (which must happen before we actually get to this app's
```

Comment:
What is a sub-assembly? Is there somewhere I can read more about this?

Ideally, this should be recorded in a README or an RFC or a design.md.